src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.CurlResponseMessage.cs

```diff
@@ -341,6 +335,7 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                     }
 
                     _easy._associatedMultiAgent.RequestUnpause(_easy);
+                    _easy._selfStrongToWeakReference.MakeStrong(); // convert from a weak to a strong ref to keep the easy alive during the read
```

Comment:
Probably a good idea.  Something would be very wrong if it wasn't, since it should be wrapping _easy, which is obviously still alive.  But an assert wouldn't hurt.